cmake/get_info.cmake

```diff
@@ -0,0 +1,310 @@
+#        GINKGO_WITH_IWYU:                       ${GINKGO_WITH_IWYU}
+#        GINKGO_CHECK_CIRCULAR_DEPS:             ${GINKGO_CHECK_CIRCULAR_DEPS}
+"
+  )
+_both(
+"#
+#     General information:
+#        CMAKE_BINARY_DIR:                       ${CMAKE_BINARY_DIR}
+#        CMAKE_SOURCE_DIR:                       ${CMAKE_SOURCE_DIR}
+#        CMAKE_CXX_COMPILER:                     ${CMAKE_CXX_COMPILER_ID} ${CMAKE_CXX_COMPILER_VERSION} on platform ${CMAKE_SYSTEM_NAME} ${CMAKE_SYSTEM_PROCESSOR}
+#                                                ${CMAKE_CXX_COMPILER}"
+  )
+
+
+
+_detailed(
+    "
+#
+#########################################################################
+#
+#   Compiled Modules
+#"
+    )
+_detailed(
+    "
+#########################################################################
+#
+#    The Core module is being compiled.
+#
+#        BUILD_SHARED_LIBS:                      ${BUILD_SHARED_LIBS}"
+    )
+IF(CMAKE_C_COMPILER_WORKS)
+    _detailed(
+        "
+#        CMAKE_C_COMPILER:                       ${CMAKE_C_COMPILER}"
+        )
+    build_type_spec(${detailed_log} "CMAKE_C_FLAGS")
+ENDIF()
+IF(CMAKE_CXX_COMPILER_WORKS)
+    _detailed(
+        "
+#        CMAKE_CXX_COMPILER:                     ${CMAKE_CXX_COMPILER}"
+        )
+    build_type_spec(${detailed_log} "CMAKE_CXX_FLAGS")
+ENDIF()
+IF(CMAKE_Fortran_COMPILER_WORKS)
+    _detailed(
+        "
+#        CMAKE_Fortran_COMPILER:                 ${CMAKE_Fortran_COMPILER}"
+        )
+ENDIF()
+_detailed(
+    "
+#        CMAKE_GENERATOR:                        ${CMAKE_GENERATOR}
+#"
+    )
+
+IF(GINKGO_BUILD_REFERENCE)
+    _detailed(
+        "
+#########################################################################
+#
+#    The Reference module is being compiled.
+#
+#        GINKGO_BUILD_REFERENCE:                 ${GINKGO_BUILD_REFERENCE}
+#        GINKGO_COMPILER_FLAGS:                  ${GINKGO_COMPILER_FLAGS}
+#
+#"
+        )
+ENDIF()
+
+
+IF(GINKGO_BUILD_OMP)
+    _detailed(
+        "
+#########################################################################
+#
+#    The OMP module is being compiled.
+#
+#    CMake related OMP variables:
+#        OpenMP_CXX_FLAGS:                        ${OpenMP_CXX_FLAGS}"
+        )
+    _detailed(
+"
+#        OpenMP_CXX_LIB_NAMES:                    ${OpenMP_CXX_LIB_NAMES}
+#        OpenMP_CXX_LIBRARIES:                    ${OpenMP_CXX_LIBRARIES}
+#
+#    Ginkgo specific OMP variables:
+#        GINKGO_COMPILER_FLAGS:                   ${GINKGO_COMPILER_FLAGS}
+#
+#"
+        )
+ENDIF()
+
+
+IF(GINKGO_BUILD_CUDA)
+    _detailed(
+        "
+#########################################################################
+#
+#    The CUDA module is being compiled.
+#
+#    CMake related CUDA variables:
+#        CMAKE_CUDA_COMPILER:                    ${CMAKE_CUDA_COMPILER}
+#        CMAKE_CUDA_COMPILER_VERSION:            ${CMAKE_CUDA_COMPILER_VERSION}"
```

Comment:
bug: I think a newline goes missing here, the output is ```CMAKE_CUDA_COMPILER_VERSION:            10.0.1#       CMAKE_CUDA_FLAGS_DEBUG                 -g```
```suggestion
#        CMAKE_CUDA_COMPILER_VERSION:            ${CMAKE_CUDA_COMPILER_VERSION}
"
```

Reply:
[It looks correct to me](https://storage.googleapis.com/gitlab-gprd-artifacts/b3/71/b371625aef4555ed2d68bdfb8a378c365272a348af12d3a1ddb806372e332781/2020_08_02/667166946/732286501/job.log?response-content-type=text%2Fplain%3B%20charset%3Dutf-8&response-content-disposition=inline&GoogleAccessId=gitlab-object-storage-prd@gitlab-production.iam.gserviceaccount.com&Signature=sWDsLwHIf9Y8%2FJ6Jy28ZfpTGUkd784cVectneksqUuvtLiZyPKmT7Bfd5eYM%0AVv39%2BQMfFHMcHhDk0oB1oQ4NB%2BE2oUZ0sbETx8iyvmxydYIEAtQ1DjTLuAnk%0Afgx2dYt%2BdcO%2BYyB%2FMZL0oNhCSljvQIIbU8p%2FgG9Telf99l%2Bt7TWd0BXAVDKX%0ARQCmoTneMaZ04GgwG1rLrPWwo5W8cHJXeuxyi%2Bl%2BcpC8U5YGUPOXQ91%2Fu3S7%0AwKtCyNfcOB0sOwR%2Fy0mnQCPfNPhzRnfz49rrCD5BSPVjrElKbTqA1xvhnZyU%0ASEtjCUM3tq5QMJm7b7KV31G01VkpCk3xWkvyFdVijg%3D%3D&Expires=1596445907). Do you get a different formatting ?